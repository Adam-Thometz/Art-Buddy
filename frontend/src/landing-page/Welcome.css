.Welcome {
  width: 85%;
  margin: 50px auto;
  border-radius: 50px;
  border: 1px solid black;
  background-color: white;
}

.Welcome-description {
  text-align: left;
  width: 90%;
  margin: 0 auto;
}