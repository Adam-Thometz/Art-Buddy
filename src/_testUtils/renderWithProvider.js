import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;