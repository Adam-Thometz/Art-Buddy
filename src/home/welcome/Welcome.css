.Welcome {
  display: flex;
  flex-direction: column;
}

.Welcome-description {
  text-align: left;
  width: 90%;
  margin: 0 auto;
}

.Welcome button {
  width: 50%;
  margin: 0 auto;
}
