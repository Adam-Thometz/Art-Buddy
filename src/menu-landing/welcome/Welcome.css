.Welcome-description {
  text-align: left;
  width: 90%;
  margin: 0 auto;
}