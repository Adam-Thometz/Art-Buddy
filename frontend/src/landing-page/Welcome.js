import React from "react";

import './Welcome.css';

import Button from "../_components/button/Button";

const Welcome = ({ setHasVisited }) => {
  const handleVisited = () => {
    setHasVisited(true);
  }
  return (
    <div className="Welcome">
      <h1>Welcome to ARTBUDDY!</h1>
      <div className="Welcome-description">
        <p>ARTBUDDY is a collection of fun and interactive fine arts games designed specifically with special education goals in mind</p>
        <p>Click below to get started</p>
        <Button outlineColor={'#89EB0A'} onClick={handleVisited}>LET'S GO!</Button>
      </div>
    </div>
  );
};

export default Welcome;