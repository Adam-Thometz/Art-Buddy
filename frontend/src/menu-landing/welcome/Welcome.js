import React from "react";

import './Welcome.css';

import Button from "../../_components/button/Button";

import colors from "../../_components/button/colorOrder";

const Welcome = ({ setHasVisited }) => {
  const handleVisited = () => {
    setHasVisited(true);
  }
  return (
    <div className="Welcome">
      <h1>WELCOME TO ARTBUDDY!</h1>
      <div className="Welcome-description">
        <p>ARTBUDDY is a collection of fun and interactive fine arts games designed specifically with special education goals in mind</p>
        <p>Click below to get started</p>
      </div>
      <Button borderColor={colors[0]} onClick={handleVisited}>LET'S GO!</Button>
    </div>
  );
};

export default Welcome;