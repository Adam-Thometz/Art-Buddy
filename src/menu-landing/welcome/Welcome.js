import React from "react";

import './Welcome.css';

import WindowNavbar from "../../_components/window-nav/WindowNavbar";
import Button from "../../_components/button/Button";

const Welcome = ({ setHasVisited }) => {
  const handleVisited = () => {
    setHasVisited(true);
  };
  
  return (
    <section className="Welcome">
      <WindowNavbar page='WELCOME TO ART BUDDY!' />
      <article className="Welcome-description">
        <p>ARTBUDDY is a collection of fun and interactive fine arts games designed specifically with special education goals in mind</p>
        <p>Click below to get started</p>
      </article>
      <Button colorId={0} onClick={handleVisited}>LET'S GO!</Button>
    </section>
  );
};

export default Welcome;