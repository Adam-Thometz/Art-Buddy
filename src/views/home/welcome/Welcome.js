import "./Welcome.css";

import WindowNavbar from "components/window-nav/WindowNavbar";
import Button from "components/button/Button";

const Welcome = ({ setHasVisited }) => {
  const handleVisited = () => setHasVisited(true);

  return (
    <section className="Welcome">
      <WindowNavbar page="WELCOME TO ART BUDDY!" />
      <article className="Welcome-description">
        <p>
          ART BUDDY is a collection of fun and interactive fine arts games
          designed specifically with academic goals in mind
        </p>
        <p>Click below to get started</p>
      </article>
      <Button colorId={0} onClick={handleVisited}>
        LET'S GO!
      </Button>
    </section>
  );
};

export default Welcome;
