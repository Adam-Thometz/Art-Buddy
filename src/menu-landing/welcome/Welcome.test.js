import React from "react";
import renderWithProvider from '../../_testUtils/renderWithProvider';
import Welcome from './Welcome'

describe('Welcome component', () => {
  it('renders without crashing', () => {
    renderWithProvider(<Welcome />);
  })

  it('matches the snapshot of the welcome page', () => {
    const { asFragment } = renderWithProvider(<Welcome />);
    expect(asFragment()).toMatchSnapshot();
  })
})