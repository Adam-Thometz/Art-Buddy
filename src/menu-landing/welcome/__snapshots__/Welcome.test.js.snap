// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome component matches the snapshot of the welcome page 1`] = `
<DocumentFragment>
  <div
    class="Welcome"
  >
    <div
      class="Welcome-description"
    >
      <p>
        ARTBUDDY is a collection of fun and interactive fine arts games designed specifically with special education goals in mind
      </p>
      <p>
        Click below to get started
      </p>
    </div>
    <button
      class="Button regular"
      style="border-color: #89eb0a;"
    >
      LET'S GO!
    </button>
  </div>
</DocumentFragment>
`;
